module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -117,126 +118,155 @@ namespace module::vision {
 
                 log<NUClear::DEBUG>(fmt::format("Found {} clusters below green horizon", clusters.size()));
 
+                // Create the Balls message, which will contain a Ball for every cluster that is a valid ball
                 auto balls = std::make_unique<Balls>();
 
-                // We still emit if there are no balls otherwise other modules are not aware that we can no longer see a
-                // ball, and they will keep using the last message sent. See head behaviour, where we use 'with' balls
-                // and react based on that last balls message.
+                // Balls is still emitted even if there are no balls, to indicate to other modules that no balls are
+                // currently visible
                 if (clusters.empty()) {
                     log<NUClear::DEBUG>("Found no balls.");
                     emit(std::move(balls));
                     return;
                 }
 
+                // Reserve the memory prematurely for efficiency
                 balls->balls.reserve(clusters.size());
 
-                balls->id        = horizon.id;
-                balls->timestamp = horizon.timestamp;
-                balls->Hcw       = horizon.Hcw;
+                balls->id        = horizon.id;         // subsumption id
+                balls->timestamp = horizon.timestamp;  // time when the image was taken
+                balls->Hcw       = horizon.Hcw;        // world to camera transform at the time the image was taken
 
+                // Check each cluster to see if it's a valid ball
                 for (auto& cluster : clusters) {
                     Ball b;
 
                     // Average the cluster to get the cones axis
                     Eigen::Vector3f axis = Eigen::Vector3f::Zero();
+                    // Add up all the unit vectors of each point (camera to point in world space) in the cluster
                     for (const auto& idx : cluster) {
                         axis += rays.col(idx);
                     }
-                    axis /= cluster.size();
+                    axis /= cluster.size();  // can this line be skipped if it's being normalised in the next line?
```

Comment:
?

Reply:
Yeah it can be skipped if it's being normalised